### DISCUSSION TITLE
Using a variable in another variable

### DISCUSSION THREAD

**Discussion by Questvba** [dbeaver/dbeaver#12228] [Q&A]
Hello,
I hope everyone is well. Thank you for welcoming me for my first post.

In sql developer, I can use this syntax which allows me to define a variable containing another variable:
```
define V1= 10
define V2= 10 + &&V1

select 
    &&V1
    ,&&V2
from dual
;
```

With DBeaver, I try in all directions but I don't get anywhere. My last attempt : 
```
@set V1 = 5
@set V2 = 10 + :V1

SELECT 
	:V1
	,:V2
FROM dual;
```

An error : **SQL Error [1008] [72000]: ORA-01008**
Can you tell me how I should proceed? Many thanks for all the information.

**Comment by Questvba** [comment]
After several tests, this seems to work 

```
@set V1 = 5
@set V2 = 10 + ${V1}

SELECT 
	:V1
	,:V2
FROM dual;
```

Can anyone tell me what the use of ${ } means?

**Reply by jmoraleda** [reply]
`${varName}` is an alternative syntax for `:varName` https://github.com/dbeaver/dbeaver/wiki/SQL-Execution#variables-binding. The `${}` syntax replaces the variable in some contexts where the the colon (:) syntax would not be replaced, such as inside strings or inside the value of other variables.

For example:

```
@set V1 = 5
@set V2 = 10 + :V1
@set V3= 10 + ${V1}
@echo "V2" equals "${V2}" and "V3" equals "${V3}"
```

prints `"V2" equals "10 + :V1" and "V3" equals "10 + 5"`
